.github/workflows/python-test.yml

```diff
@@ -225,24 +225,33 @@ jobs:
       # ----------------------------------------Huggingface Server Unit Tests------------------------------------------------
       # load cached huggingface venv if cache exists
       - name: Load cached huggingface venv
+        if: ${{ !startsWith(steps.setup-python.outputs.python-version, '3.8') }}
         id: huggingface-dependencies
         uses: actions/cache@v3
         with:
-          path: python/huggingfaceserver/.venv
+          path: /mnt/python/huggingfaceserver-venv
           key: huggingface-venv-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/kserve/poetry.lock', '**/huggingfaceserver/poetry.lock') }}
         # install huggingface server dependencies if cache does not exist
+      - name: Configure poetry
+        if: ${{ !startsWith(steps.setup-python.outputs.python-version, '3.8') }}
+        run: |
+          poetry config virtualenvs.path /mnt/python/huggingfaceserver-venv
+          poetry config virtualenvs.in-project false
       - name: Install huggingface dependencies
-        if: steps.cached-huggingface-dependencies.outputs.cache-hit != 'true'
+        if: ${{ steps.cached-huggingface-dependencies.outputs.cache-hit != 'true' && !startsWith(steps.setup-python.outputs.python-version, '3.8') }}
         run: |
+          sudo mkdir -p /mnt/python/huggingfaceserver-venv
+          # change permission so that poetry can install without sudo
+          sudo chown -R $USER /mnt/python/huggingfaceserver-venv
           cd python/huggingfaceserver
           make install_dependencies
       - name: Install huggingface server
+        if: ${{ !startsWith(steps.setup-python.outputs.python-version, '3.8') }}
         run: |
           cd python/huggingfaceserver
           make dev_install
       - name: Test huggingfaceserver
+        if: ${{ !startsWith(steps.setup-python.outputs.python-version, '3.8') }}
```

Comment:
huggingfaceserver does not support python 3.8 ? https://github.com/kserve/kserve/blob/0b2c7d9a9fa6d766adf02449f1c12d492c62a3f1/python/huggingfaceserver/pyproject.toml#L13

Reply:
Also 3.8 is eol on oct 2024